src/Components/WebAssembly/WebAssembly/src/HotReload/HotReloadAgent.cs

```diff
@@ -39,7 +39,7 @@ private void OnAssemblyLoad(object? _, AssemblyLoadEventArgs eventArgs)
             if (_deltas.TryGetValue(moduleId.Value, out var updateDeltas) && _appliedAssemblies.TryAdd(loadedAssembly, loadedAssembly))
```

Comment:
Do we use a `ConcurrentDictionary<Assembly, Assembly>` instead of a `ConcurrentBag` to protect against duplicate entries?

Reply:
We really need a ConcurrentSet here. ConcurrentBags have no lookup semantics, it gives you an item with no good way of telling you if a specific instance exists. ConcurrentDictionary with a throw away value acts as a reasonable subsitute.